x/mongo/driver/topology/topology_test.go

```diff
@@ -835,3 +835,60 @@ func serverKindFromString(t *testing.T, s string) description.ServerKind {
 
 	return description.Unknown
 }
+
+func BenchmarkSelectServerFromDescription(b *testing.B) {
+	bench := func(b *testing.B, serversHook func(servers []description.Server)) {
```

Comment:
Instead of using a closure and constructing sub-benchmarks for each case, can we iterate over the benchmark cases? Something like this:

```go
	for _, bcase := range []struct {
		name        string
		serversHook func(servers []description.Server)
	}{
		{
			"AllFit",
			func(servers []description.Server) {},
		},
		{
			"AllButOneFit",
			func(servers []description.Server) {
				servers[0].Kind = description.Unknown
			},
		},
		// Continue test cases
	} {
		bcase := bcase

		b.Run(bcase.name, func(b *testing.B) {
			// your benchmark logic
		})
	}
```

This would be more inline with out existing testing patterns.

Reply:
Changed benchmark this way